### PR TITLE
[MRG] Add checkpoints to hashval query to enable read extraction with one run

### PR DESCRIPTION
**Pull request by taylorreiter**
This PR addresses #242 point 1. It uses snakemake checkpoints to solve for the output files of rule `hashval_query`, which then produces each output for rule `extract_reads_single_hashval_file`.

Still needs to be tested :)

### CODE REVIEW

**Comment by taylorreiter** [comment]
shout out to @luizirber for helping me work through the logic of checkpoints!

**Comment by taylorreiter** [comment]
@ctb this is working on my local computer but failing CI...I don't know how to debug it. Would you be willing to take a look and see if anything pops out to you?

**Comment by taylorreiter** [comment]
It was not working on CI because i was invoking snakemake to create the directory structure that was needed for `spacegraphcats/search/query_by_hashval.py`. Thank you to @luizirber for helping me with pdb, and for teaching me all the git tricks! Ready for review and merge @ctb!

File: spacegraphcats/conf/Snakefile [hunk 1]
```diff
@@ -279,38 +279,52 @@ rule build_hashval_query_index:
...
         expand("{catlas_dir}/reads.bgz.labels", catlas_dir=catlas_dir),
-        expand("{hashval_query_dir}/{{hashval}}.cdbg_ids.txt.gz", hashval_query_dir=hashval_query_dir)
-    output:
-        join(hashval_query_dir, "{hashval}.cdbg_ids.reads.fa.gz")
+        expand("{hashval_query_dir}/contigs/{{hashval}}.cdbg_ids.txt.gz", hashval_query_dir=hashval_query_dir)
+    output: expand("{hashval_query_dir}/reads/{{hashval}}.cdbg_ids.reads.fa.gz", hashval_query_dir=hashval_query_dir)
     shell:
         "{python} -m spacegraphcats.search.extract_reads {input[0]} {input[1]} {input[2]} -o {output}"
 
+# This function will not be needed if aggregate_hashval_query works
```
Comment (ctb) [inline spacegraphcats/conf/Snakefile]: shouldn't we remove it now?

File: spacegraphcats/search/query_by_hashval.py [hunk 2]
```diff
@@ -2,6 +2,10 @@
 """
 Retrieve nodes by MinHash hashvals, using the indices created by
 spacegraphcats.cdbg.index_cdbg_by_minhash.
+Excepts as input catlas prefix, pickled hasval index, file with
```
Comment (ctb) [inline spacegraphcats/search/query_by_hashval.py]: "accepts"? Also, 'hasval' => 'hashval', 'scled' => 'scaled'.

File: spacegraphcats/conf/Snakefile (hunk 1 above)
Comment (taylorreiter) [reply spacegraphcats/conf/Snakefile]: removed!

**Comment by ctb** [comment]
LGTM!

**Pull request merged**